### PR TITLE
Taxonomy settings

### PR DESCRIPTION
**Pull request by ramirobg94**

### CODE REVIEW

File: src/arena/taxonomy/index.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
+const props = {
+  uuid: 'uuid',
+  code: 'code',
+  scientificName: 'scientificName',
+  vernacularNames: 'vernacularNames',
+  genus: 'genus',
+  family: 'family',
+};
+
+export const taxonomyVisibleFieldsOptions = [
```
Comment (SteRiccio) [inline src/arena/taxonomy/index.js]: here I see a lot of repetition.
The possible options can be an object like: 
{
    codeScientificName: [props.code, props.scientificName],
   scientificNameVernacularNames: [props.scientificName, props.vernacularNames],
   ...
}

then the "options" can be generated in the UI component (src/screens/Settings/Screens/Survey/Taxonomies/screen.js) by simply iterating over the entries of the taxonomyVisibleFieldsOptions or whatever you want to call it:
Object.entries(taxonomyVisibleFieldsOptions).map(([key, visibleFields]) => ({
   key,
   value: key,
  label: visibleFields.map(field => t(.... something using the field with internationalization )).join(', ')
})

**Review by SteRiccio — CHANGES_REQUESTED** [review]

File: src/arena/taxonomy/index.js (hunk 1 above)
Comment (ramirobg94) [reply src/arena/taxonomy/index.js]: I don't see this repetition. Here I think that it is better to be explicit.

File: src/arena/taxonomy/index.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    label: '(Code), ScientificName, vernacularNames',
+  },
+  {
+    key: `${props.scientificName}.${props.vernacularNames}`,
+    value: [props.scientificName, props.vernacularNames],
+    label: 'ScientificName, vernacularNames',
+  },
+];
+
+export const exampleTaxonomy = {
```
Comment (SteRiccio) [inline src/arena/taxonomy/index.js]: this one should be an exampleTaxon, not taxonomy...

Comment (ramirobg94) [reply src/arena/taxonomy/index.js]: Fixed,

File: src/arena/taxonomy/index.js (hunk 1 above)
Comment (SteRiccio) [reply src/arena/taxonomy/index.js]: Do whatever you want then

Comment (SteRiccio) [reply src/arena/taxonomy/index.js]: I was talking about the the entire taxonomyVisibleFieldsOptions object.
I mean that for every option the thing that you are considering is the fields and their order, for example [props.code, props.genus, props.family, props.scientificName]. When you build the options items, you can derive key and label properties from the value, so in the end it's better to define only an object with only the possible combinations, like:
{
codeScientificName: [props.code, props.scientificName],
scientificNameVernacularNames: [props.scientificName, props.vernacularNames],
...
}
and generate an "options" object derived by the one above... is it clear now?

Comment (ramirobg94) [reply src/arena/taxonomy/index.js]: Please check the changes: https://github.com/openforis/arena-mobile/pull/415/commits/f231312eeb5e6848aede3934d938da51b73f7d44


In the future, I will be grateful if expressions such as "Is it clear now", "..." or "?!"; or categorical affirmations like "always", "never" or "a lot of" are avoided. These expressions could sound aggressive and inquisitive.

File: src/screens/Settings/Screens/Survey/Taxonomies/screen.js [hunk 3]
```diff
@@ -61,47 +62,62 @@ const SettingsSurveyTaxonomies = () => {
...
-                  style={[
-                    styles.optionText,
-                    taxonomyVisibleFields === option.key &&
-                      styles.optionTextSelected,
-                  ]}>
-                  {option.label}
-                </Text>
-              </TouchableOpacity>
-            ))}
+            {Object.entries(taxonomyVisibleFieldsOptions)?.map(
```
Comment (SteRiccio) [inline src/screens/Settings/Screens/Survey/Taxonomies/screen.js]: Object.entries always returns an array... no need for the optional chaining operator ?.

File: src/screens/Settings/Screens/Survey/Taxonomies/screen.js [hunk 4]
```diff
@@ -61,47 +62,62 @@ const SettingsSurveyTaxonomies = () => {
...
+              (option, index) => {
+                console.log(option);
```
Comment (SteRiccio) [inline src/screens/Settings/Screens/Survey/Taxonomies/screen.js]: please remove the console.log

File: src/screens/Settings/Screens/Survey/Taxonomies/screen.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+            <Text style={baseStyles.textStyle.info}>
+              {t('Settings:survey.taxonomies.screen.description')}
+            </Text>
+            <Text style={baseStyles.textStyle.info}>
+              {t('Settings:survey.taxonomies.screen.info')}
+            </Text>
+
+            {Object.entries(taxonomyVisibleFieldsOptions)?.map(
+              (option, index) => {
+                console.log(option);
```
Comment (ramirobg94) [inline src/screens/Settings/Screens/Survey/Taxonomies/screen.js]:
```suggestion
```

File: src/screens/Settings/Screens/Survey/Taxonomies/screen.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+              {t('Settings:survey.taxonomies.screen.header')}
+            </Text>
+            <Text style={baseStyles.textStyle.info}>
+              {t('Settings:survey.taxonomies.screen.description')}
+            </Text>
+            <Text style={baseStyles.textStyle.info}>
+              {t('Settings:survey.taxonomies.screen.info')}
+            </Text>
+
+            {Object.entries(taxonomyVisibleFieldsOptions)?.map(
```
Comment (ramirobg94) [inline src/screens/Settings/Screens/Survey/Taxonomies/screen.js]:
```suggestion
            {Object.entries(taxonomyVisibleFieldsOptions).map(
```

File: src/i18n/en/Settings.json [hunk 7]
```diff
@@ -14,8 +14,25 @@
...
+      "screen": {
+        "header": "Taxonomies",
+        "description": "Select how do you wnat to see the taxonomies by default",
+        "info": "This value is going to be overwritten by the attribute configuration",
+        "fields": {
+          "code": "(Code)",
+          "genus": "Genus",
+          "family": "Family",
+          "scientificName": "ScientificName",
+          "vernacularNames": "VernacularNames"
```
Comment (ramirobg94) [inline src/i18n/en/Settings.json]:
```suggestion
          "scientificName": "Scientific Name",
          "vernacularNames": "Vernacular Names"
```

File: src/arena/taxonomy/index.js (hunk 1 above)
Comment (SteRiccio) [reply src/arena/taxonomy/index.js]: noted

**Review by SteRiccio — APPROVED** [review]

**Pull request merged**